### PR TITLE
Move schedule name into `Schedule`

### LINKED ISSUE #9510: Schedule build errors should include the schedule in which the problem occurred when reporting problems
## What problem does this solve or what need does it fill?

When errors are reported during schedule building, the schedule name is not included. 

It would be helpful to be able to track down the exact source of the problem, especially since that information.

## What solution would you like?

Currently:

```
2023-08-20T17:23:10.908705Z  WARN bevy_ecs::schedule::schedule: 1 pairs of systems with conflicting data access have indeterminate execution order. Consider adding `before`, `after`, or `ambiguous_with` relationships between these:
 -- update_action_state<PlayerAction> (in set Update) and ui_focus_system (in set Focus)
    conflict on: ["bevy_ui::focus::Interaction"]
```

In the preamble to this reporting, we should report the schedule name if it is available.

## Additional context

We won't always be able to report the schedule name, since this information is stored in `Schedules`, not on each individual `Schedule`. In theory I suppose we could change that though.

### PR DESCRIPTION
**Pull request by hymm**
# Objective

- Move schedule name into `Schedule` to allow the schedule name to be used for errors and tracing in Schedule methods
- Fixes #9510

## Solution

- Move label onto `Schedule` and adjust api's on `World` and `Schedule` to not pass explicit label where it makes sense to.
- add name to errors and tracing.
- `Schedule::new` now takes a label so either add the label or use `Schedule::default` which uses a default label. `default` is mostly used in doc examples and tests.

---

## Changelog

- move label onto `Schedule` to improve error message and logging for schedules.

## Migration Guide

`Schedule::new` and `App::add_schedule`
```rust
// old
let schedule = Schedule::new();
app.add_schedule(MyLabel, schedule);

// new
let schedule = Schedule::new(MyLabel);
app.add_schedule(schedule);
```

if you aren't using a label and are using the schedule struct directly you can use the default constructor.
```rust
// old
let schedule = Schedule::new();
schedule.run(world);

// new
let schedule = Schedule::default();
schedule.run(world);
```

`Schedules:insert`
```rust
// old
let schedule = Schedule::new();
schedules.insert(MyLabel, schedule);

// new
let schedule = Schedule::new(MyLabel);
schedules.insert(schedule);
```

`World::add_schedule`
```rust
// old
let schedule = Schedule::new();
world.add_schedule(MyLabel, schedule);

// new
let schedule = Schedule::new(MyLabel);
world.add_schedule(schedule);
```

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Very straightforward, but extremely useful for diagnostics.

File: crates/bevy_ecs/src/world/mod.rs
Comment (JoJoJet) [inline crates/bevy_ecs/src/world/mod.rs:1775]: Should `remove_entry` still return the label?

Comment (hymm) [reply crates/bevy_ecs/src/world/mod.rs:1775]: I'm not sure if `remove_entry` has any use. We already have `remove`, so we should just delete `remove_entry` if it doesn't. It's slightly useful in that you can get the owned Boxed label for free, but not sure that's worth the added api. 

I should switch to using `remove` here, since we're not using the label.

Comment (JoJoJet) [reply crates/bevy_ecs/src/world/mod.rs]: Fair enough. If #7762 gets merged then we should definitely get rid of `remove_entry`, since there'd be no benefit to keeping the schedule label.

File: crates/bevy_ecs/src/schedule/schedule.rs
Comment (JoJoJet) [inline crates/bevy_ecs/src/schedule/schedule.rs:169]: Will this documentation be visible to users?

**Review by JoJoJet — APPROVED** [review]
I have one concern about docs, but otherwise LGTM.

File: crates/bevy_ecs/src/schedule/schedule.rs
Comment (alice-i-cecile) [reply crates/bevy_ecs/src/schedule/schedule.rs:169]: It should be yes. I'll verify before merging though.

**Comment by alice-i-cecile** [comment]
image

The docs on the trait impl show up fine :)

**Comment by alice-i-cecile** [comment]
Looks like merge conflict-y issues: can you rebase and fix this up?

**Comment by alice-i-cecile** [comment]
@hymm CI is still failing: looks like another use of `Schedule` was added.

**Comment by JoJoJet** [comment]
> image
> 
> The docs on the trait impl show up fine :)

They show up, but I don't think they'll show up where anyone will see them. It'd have to be on the `default` fn to be useful, I think.

**Pull request merged**
